src/main/java/com/yugabyte/sample/apps/SqlWarehouseStock.java

```diff
@@ -0,0 +1,154 @@
+ * b. Restock: Restocks items whose stock has fallen below 10 items by
+ *   adding 100 items.
+ *
+ * NewOrder Operation:
+ * 1. Pick a random item_id.
+ * 2. Decrement the stock of the item only when there is enough stock.
+ *
+ * Restock Operation:
+ * 1. Scan the table, restock by 100 when below 10.
+ */
+public class SqlWarehouseStock extends AppBase {
+  private static final Logger LOG = Logger.getLogger(SqlWarehouseStock.class);
+
+  // Static initialization of this app's config.
+  static {
+    // Use 1 Restock thread and 100 NewOrder threads.
+    appConfig.readIOPSPercentage = -1;
+    appConfig.numReaderThreads = 1;
+    appConfig.numWriterThreads = 100;
+    // Disable number of keys.
+    appConfig.numKeysToRead = -1;
+    appConfig.numKeysToWrite = -1;
+    // Run the app for 1 minute.
+    appConfig.runTimeSeconds = 60;
+    // Report restart read requests metric by default.
+    appConfig.restartReadsReported = true;
+    // Avoid load balancing errors.
+    appConfig.loadBalance = false;
+    appConfig.disableYBLoadBalancingPolicy = true;
+  }
+
+  // The default table name to create and use for ops.
+  private static final String DEFAULT_TABLE_NAME = "warehouse_stock";
+
+  // The number of items in the warehouse.
+  private static final int NUM_ITEMS = 1000;
+
+  // The stock level below which restocking is needed.
+  private static final int RESTOCK_THRESHOLD = 10;
+
+  // The amount to restock by.
+  private static final int RESTOCK_AMOUNT = 100;
+
+  // Initial stock.
+  private static final int INITIAL_STOCK = 100;
+
+  @Override
+  public void createTablesIfNeeded(TableOp tableOp) throws Exception {
+    try (Connection connection = getPostgresConnection()) {
+      // Every run should start cleanly.
+      connection.createStatement().execute(
+          String.format("DROP TABLE IF EXISTS %s", getTableName()));
+      LOG.info("Dropping any table(s) left from previous runs if any");
+      connection.createStatement().execute(String.format(
+          "CREATE TABLE %s (item_id INT PRIMARY KEY, stock INT)" +
+          " SPLIT INTO 24 TABLETS",
+          getTableName()));
+      LOG.info(String.format("Created table: %s", getTableName()));
+      int numRows = connection.createStatement().executeUpdate(String.format(
+          "INSERT INTO %s SELECT GENERATE_SERIES(0, %d-1), %d",
+          getTableName(), NUM_ITEMS, INITIAL_STOCK));
+      LOG.info(String.format(
+          "Inserted %d rows into %s", numRows, getTableName()));
+    }
+  }
+
+  @Override
+  public String getTableName() {
+    String tableName = appConfig.tableName != null ?
+        appConfig.tableName : DEFAULT_TABLE_NAME;
+    return tableName.toLowerCase();
+  }
+
+  // Executes the Restock operation.
+  @Override
+  public long doRead() {
+    try (Connection connection = getPostgresConnection();
+         Statement statement = connection.createStatement()) {
+      statement.executeUpdate(String.format(
```

Comment:
Maybe in `doRead` workload do a read and check consistency? And this logic can be moved in `doWrite` method and activate on a coin flip?

Reply:
I check if there are any stale reads now. The name doRead is unfortunate but it is better to report latency and throughput separately for different operations.